src/core/ddsc/include/dds/ddsc/dds_opcodes.h

```diff
@@ -236,9 +238,10 @@ enum dds_stream_typecode_subtype {
    and (3) the discriminator can be an integral type (or enumerated - here treated as equivalent).
    What it can't be is a floating-point type. So DEF and FP need never be set at the same time.
    There are only a few flag bits, so saving one is not such a bad idea. */
-#define DDS_OP_FLAG_FP  (1u << 1) /* floating-point: applicable to {4,8}BY and arrays, sequences of them */
-#define DDS_OP_FLAG_SGN (1u << 2) /* signed: applicable to {1,2,4,8}BY and arrays, sequences of them */
-#define DDS_OP_FLAG_MU  (1u << 3) /* must-understand flag, used with PLM in parameter list CDR */
+#define DDS_OP_FLAG_FP   (1u << 1) /* floating-point: applicable to {4,8}BY and arrays, sequences of them */
+#define DDS_OP_FLAG_SGN  (1u << 2) /* signed: applicable to {1,2,4,8}BY and arrays, sequences of them */
+#define DDS_OP_FLAG_MU   (1u << 3) /* must-understand flag */
+#define DDS_OP_FLAG_BASE (1u << 4) /* jump to base type, used with PLM in PL-CDR */
```

Comment:
I am ok with solving this problem by supporting fragmented member tables instead of inlining everything, at the very least because the alternative requires more changes to the IDL compiler. I do wonder why you chose to introduce a new flag on the `PLM` instruction, it seems to me like allowing a member table to include `JSR`s would do just as well. If that really is the case, I would find it more elegant, but I may be overlooking something.

(Changing it appears to be very little work — but again, I may be overlooking something!)

Reply:
A reason for introducing this flag and not using JSR is that the code that handles the PLM list (e.g. skipping to end of the list in the serializer when reading PL-CDR) remains simple and does not need to change, it can safely do `ops += 2` and assume every 2nd op is a PLM or RTS. The same in IDLC, e.g. the `shift_plm_list_offsets` function. Although this could also be solved by adding a dummy (0) instruction after the JSR of course, and you're right that the changes required for using JSR in the list are straightforward. 

Another use of this flag is skipping the DHEADER for inheritance of appendable structs. When the flag is set on the ADR | EXT opcode, this instructs the serializer to skip the DLC instruction and continue adding the members from the base type as part of the current (derived) type (implemented in 69f4657)

Using the flag for both cases improves the readability of the generated opcodes imho, which can be useful for debugging, and there are not really drawbacks, except that it occupies an extra flag bit.

Reply:
That makes sense! Thanks for explaining.